rust/datafusion/src/logical_plan/mod.rs

```diff
@@ -21,2300 +21,21 @@
-pub fn max(expr: Expr) -> Expr {
-    Expr::AggregateFunction {
-        fun: aggregates::AggregateFunction::Max,
-        distinct: false,
-        args: vec![expr],
-    }
-}
-
-/// Create an expression to represent the sum() aggregate function
-pub fn sum(expr: Expr) -> Expr {
-    Expr::AggregateFunction {
-        fun: aggregates::AggregateFunction::Sum,
-        distinct: false,
-        args: vec![expr],
-    }
-}
-
-/// Create an expression to represent the avg() aggregate function
-pub fn avg(expr: Expr) -> Expr {
-    Expr::AggregateFunction {
-        fun: aggregates::AggregateFunction::Avg,
-        distinct: false,
-        args: vec![expr],
-    }
-}
-
-/// Create an expression to represent the count() aggregate function
-pub fn count(expr: Expr) -> Expr {
-    Expr::AggregateFunction {
-        fun: aggregates::AggregateFunction::Count,
-        distinct: false,
-        args: vec![expr],
-    }
-}
-
-/// Whether it can be represented as a literal expression
-pub trait Literal {
-    /// convert the value to a Literal expression
-    fn lit(&self) -> Expr;
-}
-
-impl Literal for &str {
-    fn lit(&self) -> Expr {
-        Expr::Literal(ScalarValue::Utf8(Some((*self).to_owned())))
-    }
-}
-
-impl Literal for String {
-    fn lit(&self) -> Expr {
-        Expr::Literal(ScalarValue::Utf8(Some((*self).to_owned())))
-    }
-}
-
-macro_rules! make_literal {
-    ($TYPE:ty, $SCALAR:ident) => {
-        #[allow(missing_docs)]
-        impl Literal for $TYPE {
-            fn lit(&self) -> Expr {
-                Expr::Literal(ScalarValue::$SCALAR(Some(self.clone())))
-            }
-        }
-    };
-}
-
-make_literal!(bool, Boolean);
-make_literal!(f32, Float32);
-make_literal!(f64, Float64);
-make_literal!(i8, Int8);
-make_literal!(i16, Int16);
-make_literal!(i32, Int32);
-make_literal!(i64, Int64);
-make_literal!(u8, UInt8);
-make_literal!(u16, UInt16);
-make_literal!(u32, UInt32);
-make_literal!(u64, UInt64);
-
-/// Create a literal expression
-pub fn lit<T: Literal>(n: T) -> Expr {
-    n.lit()
-}
-
-/// Create an convenience function representing a unary scalar function
-macro_rules! unary_math_expr {
-    ($ENUM:ident, $FUNC:ident) => {
-        #[allow(missing_docs)]
-        pub fn $FUNC(e: Expr) -> Expr {
-            Expr::ScalarFunction {
-                fun: functions::BuiltinScalarFunction::$ENUM,
-                args: vec![e],
-            }
-        }
-    };
-}
-
-// generate methods for creating the supported unary math expressions
-unary_math_expr!(Sqrt, sqrt);
-unary_math_expr!(Sin, sin);
-unary_math_expr!(Cos, cos);
-unary_math_expr!(Tan, tan);
-unary_math_expr!(Asin, asin);
-unary_math_expr!(Acos, acos);
-unary_math_expr!(Atan, atan);
-unary_math_expr!(Floor, floor);
-unary_math_expr!(Ceil, ceil);
-unary_math_expr!(Round, round);
-unary_math_expr!(Trunc, trunc);
-unary_math_expr!(Abs, abs);
-unary_math_expr!(Signum, signum);
-unary_math_expr!(Exp, exp);
-unary_math_expr!(Log, ln);
-unary_math_expr!(Log2, log2);
```

Comment:
Should these be removed? I though that they were needed to plan these expressions.

Reply:
The compiler started complaining to me that they were not used so I removed them and the tests keep passing

I just tested with the repl and I seem to be able to plan the expressions just fine, so I guess the compiler was right from what I can see. Maybe I am missing something


```
(arrow_dev) alamb@MacBook-Pro:~/Software/arrow2/rust$ git status
On branch alamb/ARROW-10559-split-up
Your branch is up to date with 'alamb/alamb/ARROW-10559-split-up'.

nothing to commit, working tree clean
(arrow_dev) alamb@MacBook-Pro:~/Software/arrow2/rust$ cd datafusion/
(arrow_dev) alamb@MacBook-Pro:~/Software/arrow2/rust/datafusion$ echo "1" > /tmp/table.csv
(arrow_dev) alamb@MacBook-Pro:~/Software/arrow2/rust/datafusion$ echo "2" >> /tmp/table.csv
(arrow_dev) alamb@MacBook-Pro:~/Software/arrow2/rust/datafusion$ 
(arrow_dev) alamb@MacBook-Pro:~/Software/arrow2/rust/datafusion$ cargo run --bin repl
    Finished dev [unoptimized + debuginfo] target(s) in 0.13s
     Running `/Users/alamb/Software/arrow2/rust/target/debug/repl`
> 

CREATE EXTERNAL TABLE t(a int)
STORED AS CSV
LOCATION '/tmp/table.csv';

0 rows in set. Query took 0 seconds.
> select sqrt(a) from t;
+--------------------+
| sqrt(a)            |
+--------------------+
| 1                  |
| 1.4142135623730951 |
+--------------------+
2 row in set. Query took 0 seconds.
> 
```

🤔  So I guess they aren't needed?

Reply:
I think that they are used when creating a statement via the DataFrame API. I.e. in `examples/dataframe.rs`, if we want to create an expression that is `sqrt(col("c1"))`, we need to import `sqrt` from `logical_plan`. No?

I find it odd that the compiler is arguing that `pub` functions are not being used, though xD

Reply:
I think the issue is that I didn't then publically export them. I'll put the code back in.

It is probably bad that we didn't have tests that failed when they got removed

Reply:
Code restored in https://github.com/apache/arrow/pull/8639/commits/ed05ee56cd99d8e22c5f0f3aec8ecddcd22bd1e0 and publically `use`d in https://github.com/apache/arrow/pull/8639/commits/60cd9fe463bb79db24f221a354a1f3334aa577d7